applications/tari_dan_wallet_daemon/src/services/account_monitor.rs

```diff
@@ -227,6 +227,10 @@ where
 
         let balance = vault.balance();
         let vault_addr = SubstateAddress::Vault(*vault.vault_id());
+        if accounts_api.get_account_by_address(account_addr).is_err() {
```

Comment:
```suggestion
        if accounts_api.get_account_by_address(account_addr).optional()?.is_none() {
```

Reply:
or create `accounts_api.exists_by_address(account_addr)?` and internally either create the `tx.accounts_exists_by_address` call or use `accounts_api.get_account_by_address(account_addr).optional()?.is_none()` inside the function.

This is so that if we want to change the backend for these calls we don't have to change usage everywhere.

Reply:
done